src/modelldcatnotordf/modelldcatno.py

```diff
@@ -500,3 +513,121 @@ def _has_type_to_graph(self, _self: Any) -> None:
                     self._g.add((_has_type, p, o))
 
                 self._g.add((_self, MODELLDCATNO.hasType, _has_type,))
+
+
+class Role(ModelProperty):
+    """A class representing a modelldcatno:Role."""
+
+    __slots__ = (
+        "_identifier",
+        "_has_object_type",
+    )
+
+    _identifier: URI
+    _has_object_type: ObjectType
+    _g: Graph
+
+    def __init__(self) -> None:
+        """Inits an object with default values."""
+        super().__init__()
+
+    @property
+    def has_object_type(self: Role) -> ObjectType:
+        """Get/set for has_object_type."""
+        return self._has_object_type
+
+    @has_object_type.setter
+    def has_object_type(self: Role, has_object_type: Any) -> None:
+        self._has_object_type = has_object_type
+
+    def to_rdf(
+        self: Role, format: str = "turtle", encoding: Optional[str] = "utf-8"
+    ) -> str:
+        """Maps the role to rdf.
+
+        Args:
+            format: a valid format. Default: turtle
+            encoding: the encoding to serialize into
+
+        Returns:
+            a rdf serialization as a string according to format.
+        """
+        return self._to_graph().serialize(format=format, encoding=encoding)
+
+    def _to_graph(
+        self: Role, type: str = MODELLDCATNO.Role, selfobject: Any = None
+    ) -> Graph:
+        """Returns the role as graph.
+
+        Args:
+            type: type for identifying class. Default: MODELLDCATNO.Role
+            selfobject: an bnode or URI passed from an subclass Default: None
+
+        Returns:
+            the role graph
+        """
+        if getattr(self, "identifier", None):
+            _self = URIRef(self.identifier)
+        else:
+            _self = BNode()
```

Comment:
Skal `selfobject` tas i bruk her som i den andre klassen? Hvis ikke, så kan hele statement skrives som en deklarasjon

Reply:
Ja, _self lages her, og brukes som argument til superklassen for å få riktig type på trippelet. Og så brukes det senere i subklassen når man skal lage properties. Poenget med å lage selfobjektet bare en gang når det subklassing av klasser som også kan være instanser er at hvis ikke legger rdflib til en dobbel trippel, en ny trippel for hver gang man kalles BNode